ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/attestation/ValidateableAttestation.java

```diff
@@ -38,6 +38,7 @@ public class ValidateableAttestation {
   private final Supplier<Bytes32> hashTreeRoot;
   private final AtomicBoolean gossiped = new AtomicBoolean(false);
   private final boolean producedLocally;
+  private final boolean producedFromNonCanonicalBlock;
```

Comment:
something like `alreadyValidated` or `previouslyValidated` or similar - looking for 'what' the context is, less than 'why' (from a non canonical block)

This will allow us to re-use it, should the need arise...

Reply:
Yeah definitely agree. Will change it across, which will cover the other points as well.

Reply:
I'd say `signatureValidated`.  The validation is in a particular context - it's still important to check that we have the same `committeeShufflingSeed` etc or the attestation may wind up invalid when included in a different block.